webclient/arch_z/views.py

```diff
@@ -835,35 +841,40 @@ def vratit(request, ident_cely):
         if form.is_valid():
             fedora_trasnaction = az.create_transaction(request.user)
             try:
-                if az.lokalita and az.stav == AZ_STAV_ARCHIVOVANY:
-                    az.lokalita.igsn_hide()
-            except ObjectDoesNotExist:
-                pass
-            duvod = form.cleaned_data["reason"]
-            projekt = None
-            if az.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_AKCE:
-                projekt = az.akce.projekt
-            # BR-A-3
-            if az.stav == AZ_STAV_ODESLANY and projekt is not None:
-                #  Return also project from the states P6 or P5 to P4
-                projekt.active_transaction = fedora_trasnaction
-                projekt_stav = projekt.stav
-                logger.debug("arch_z.views.vratit.valid", extra={"ident_cely": ident_cely, "stav": projekt.stav})
-                if projekt_stav == PROJEKT_STAV_UZAVRENY:
-                    projekt.set_vracen(request.user, projekt_stav - 1, "Automatické vrácení projektu")
-                    projekt.save()
-                if projekt_stav == PROJEKT_STAV_ARCHIVOVANY:
-                    projekt.set_vracen(request.user, projekt_stav - 1, "Automatické vrácení projektu")
-                    projekt.save()
-                    projekt.set_vracen(request.user, projekt_stav - 2, "Automatické vrácení projektu")
-                    projekt.save()
-            before_save_state = az.stav
-            az.set_vraceny(request.user, az.stav - 1, duvod)
-            az.close_active_transaction_when_finished = True
-            az.save()
-            if before_save_state == AZ_STAV_ODESLANY:
-                Mailer.send_ev01(zaznam=az, reason=duvod)
-            fedora_trasnaction.success_message = get_message(az, "USPESNE_VRACENA")
+                try:
+                    if az.lokalita and az.stav == AZ_STAV_ARCHIVOVANY:
+                        az.lokalita.igsn_hide()
```

Comment:
Co se stane, když se toto povede a zbytek ne? Nezůstane to tak, že v DataCite to bude už skryté, ale záznam samotný se rollbackne? Nemělo by to být v opačném pořadí?